### PR TITLE
Add sorting function to Saved Rules Selector

### PR DESCRIPTION
**Pull request by assuntad23**
Adding sorting to SavedRuleSelector as requested in #13192 to improve usefulness/ease of finding rules

## How to test the changes?
(Select all options that apply)
- [X] Instructions for manual testing are as follows:
  1. Create a few sets of rules in the Rule Builder (you can use the [Rule Based Uploader tutorial](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html), to make it easy)
  2. Try to space them out a bit, so that you can see that the dropdown from the Saved Rule Selector sorts them starting with the most recent.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by jmchilton** [comment]
The linting issues seem legitimate. You may want to copy the input sorting rules into a data parameter before sorting them so you aren't mutating an input property.

**Comment by assuntad23** [comment]
> 

Using a data parameter still causes a linting issue, "Unexpected side effect in [...] computed property", so I'm just using a local variable. Thanks for taking a look at this! The linting should be fine now.

File: client/src/components/RuleBuilder/SavedRulesSelector.vue [hunk 1]
```diff
@@ -42,11 +42,20 @@ export default {
         numOfSavedRules: function () {
             return this.savedRules.length;
         },
+        sortSavedRules: function () {
+            var sortedRules = this.savedRules;
+            return sortedRules.sort(this.onSessionDateTime);
```
Comment (mvdbeek) [inline client/src/components/RuleBuilder/SavedRulesSelector.vue]: That's still sorting the prop in place, isn't it (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) ?
Screenshot 2022-02-23 at 11 04 47

Compare that with

Screenshot 2022-02-23 at 11 06 11



```suggestion
            return [...this.savedRules].sort(this.onSessionDateTime);
```
is my totally untested suggestion

Comment (assuntad23) [reply client/src/components/RuleBuilder/SavedRulesSelector.vue]: I tested it! 👍🏻

**Comment by mvdbeek** [comment]
very cool, thank you @assuntad23!

**Pull request merged**